docs: Update README.md to include posthog details (#3389)
mitodl/mitxpro: README.md

We are using PostHog for managing the features. PostHog provides many built-in filters/conditions to enable/disable features in the application without any code change or deployment.

> _NOTE:_ We are using [olposthog](https://github.com/mitodl/ol-django/tree/main/src/olposthog) which is our own wrapper around PostHog to make things simpler and add caching. As part of this, we run `./manage.py createcachetable` to create the cache table for the feature flags. So that you know, your application would not work if you haven't created the cache table.

Reviewer: ```suggestion
_NOTE:_ We are using [olposthog](https://github.com/mitodl/ol-django/tree/main/src/olposthog) which is our own wrapper around PostHog to make things simpler and add caching. As part of this, we run `./manage.py createcachetable` to create the cache table for the feature flags. Just so you know, your application will not work if you haven't created the cache table.
```

- POSTHOG_FEATURE_FLAG_REQUEST_TIMEOUT_MS (`Default value: 3000`)
- POSTHOG_MAX_RETRIES (`Default value: 3`)

# PostHog Feature Flags

> We are using a couple of feature flags for xPRO. All these feature flags are listed in [features.py](https://github.com/mitodl/mitxpro/blob/master/mitxpro/features.py). To work with these flags you need to integrate Posthog as mentioned in the above step. If you don't have the API Key or you do not want to integrate the Posthog but still want to use it, you can use the default return value in the `is_enabled` method which is called for accessing the value of a feature flag.

Reviewer: ```suggestion
We are using a couple of feature flags for xPRO. All these feature flags are listed in [features.py](https://github.com/mitodl/mitxpro/blob/master/mitxpro/features.py). To work with these flags, you need to integrate Posthog as mentioned in the step above. If you don't have the API key or you do not want to integrate Posthog but still want to use it, you can use the default return value in the `is_enabled` method which is called for accessing the value of a feature flag.
```
